### ISSUE TITLE
[bug]: Invalid Config + Config Spec results in FE failing in a non-communicative way

### ISSUE THREAD

**Issue by ProofOfKeags** [Start9Labs/embassy-os#1374]
### Prerequisites

- [X] I have searched for [existing issues](https://github.com/start9labs/embassy-os/issues) that already report this problem, without success.

### EmbassyOS Version

0.3.0

### Device

Laptop/Desktop

### Device OS

Linux

### Device OS Version

Ubuntu 21.04

### Browser

Brave

### Browser Version

1.19.92

### Current Behavior

If ever a situation arises where a config entry doesn't match its corresponding spec (structurally) i.e. finds a string where an object should be, it will not allow you to save it. However because it is a *structural* failing (which is not permitted in the UI itself) the code does not exist for indicating the error in the modal, nor giving the means to correct it. This can result in deeply nested configs appearing to be fine but failing to save and no error message being rendered.

### Expected Behavior

Bad packages should never cause the OS to enter "undefined behavior".

If it is the case that the config received doesn't validate against its own spec, at minimum we need to identify this and give a top level error, indicating to the user that attempting to save the config in its current state is a futile effort.

One step better than that would be resetting the config to its defaults with user approval. 

One step better than that would be to zero in on only the subset of the config that is broken, and reset *that* to defaults with user approval.

### Steps to Reproduce

1. Install a package whose config doesn't validate against its own spec in the config get action
2. click configure
3. attempt to save as is
4. observe nothing happens

**Comment by ProofOfKeags** [comment]
CC @MattDHill

**Comment by chrisguida** [comment]
>One step better than that would be to zero in on only the subset of the config that is broken, and reset that to defaults with user approval.

This would be awesome

**Comment by chrisguida** [comment]
Btw the validation error is displayed, it's just hidden if you don't happen to be looking at it

<img width="722" alt="Screen Shot 2022-04-01 at 10 35 06 AM" src="https://user-images.githubusercontent.com/7445670/161305425-e395882d-7da6-4e48-ab6c-acadaed922a0.png">

Forcing validation errors to always be visible could go a long way to ameliorating this issue

**Comment by chrisguida** [comment]
Another idea would be to spit validation errors out to the console

**Comment by ProofOfKeags** [comment]
> Another idea would be to spit validation errors out to the console

This only helps debug, it gives no means of remedying that particular error without resetting the whole config back to its defaults.

**Comment by chrisguida** [comment]
It at least tells the dev what the problem with the new config is

**Comment by ProofOfKeags** [comment]
certainly. But this ticket is about the user, not the dev. If you want to create a second ticket for dev improvements to reduce the probability of this occuring we can, but we have to protect against malformed packages in the OS irrespective of the dev tools used.

**Comment by chrisguida** [comment]
True, users may install malformed packages in the future. If we fix for users, that should suffice to fix for devs as well.

**Comment by elvece** [comment]
I think this was recently addressed by @MattDHill in [cleanup/031](https://github.com/Start9Labs/embassy-os/pull/1558) - will confirm in testing.
